free-1.md

```diff
@@ -1,68 +1,198 @@
+between the user and the computer. It would be awesome to allow for users to
+express what they would like to draw in a language and syntax that feels
+natural to them, while also getting visual feedback about what they're drawing
+and what they could choose to do next.
 
 ### Why a language?
-_Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+I think that I should address this question in two parts:
+   1. This question should be solved with a DSL because geometric writing is
+   already designed to unambiguously communicate geometric constructions. There
+   we should try to teach computers how to understand that language, or
+   something similar to it.
+   1. Why interactive? Is making something interactive really part of DSLs? Is
+   making something interactive really changing the language? I argue yes.
+   Language is fundamentally a framework for communication. To me it seems odd
+   that programming languages so far have just been about users telling
+   computers what to do. Allowing for computers to give feedback to users
+   expands the language to allow for 2-way communication. And the way the
+   computer sends back information will need to be designed well to make it
+   most useful to the users.
 
 ### Why you?
-_What excites you about this idea? How did you come up with it?_
 
+I've always enjoyed geometry, and am excited about building an interactive
+system which makes computers a better platform for exploring mathematical ideas
+that are not fully formed yet.
 
 ### Domain
-_Describe the project's domain in five words._
 
+Interactively creating geometric drawings
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
 
+Users could interact with the language statically or interactively.
+
+In either case they would write statements like:
+
+```
+circle x with center C
+line y tangent to C at point D
+line z through points C and D
+```
+
+I really like language idea because it mirrors the way that mathematicians
+write about geometry. Who hasn't seen something like:
+
+Let x be a circle centered at C. Let y be a line tangent to x at point D. Let z
+be a line through C and D. ...more proof...
 
 ### Operation (semantics)
-_What might happen when a program runs? How does a program interact with the
-user? What kinds of errors might occur, and how might they be communicated to
-the user?_
 
+So operation would work in two ways.
+
+The first of these is the static mode. People write stuff like above and
+trigger compilation which generates an image or LaTeX code, or something like
+that.
+
+The more interesting would be interactive mode, where as the users enter each
+line the image is drawn for them to see what they have, as well as to indicate
+potentially interesting thing to do next (IE, highlighting the tangent point
```

Comment:
Yeah - something like that!

I also think imagining it as a very eager Geometry IDE wouldn't be a bad analogy either